virttest/utils_test/libvirt.py

```diff
@@ -3168,3 +3170,90 @@ def create_secret(params):
         raise exceptions.TestError("Fail to get newly created secret uuid")
 
     return sec_uuid
+
+
+def setup_libvirtd_conf_dict(params):
+    """
+    Read and set the required parameters to dict
+
+    :param params: the parameters to be used
+    :return: a dict that includes required parameters
+    """
+    conf_dict = {}
+
+    conf_dict['keepalive_interval'] = int(params.get("keepalive_interval", '5'))
+    conf_dict['log_level'] = int(params.get("log_level", '3'))
+    conf_dict['log_outputs'] = '"%s:file:%s"' % (params.get("log_level", '3'),
+                                                 params.get("libvirt_log",
+                                                 "/var/log/libvirt/libvirtd.log"))
+    logging.debug("Assemble libvirtd configuration dict as below:%s\n",
+                  conf_dict)
+    return conf_dict
+
+
+def config_or_recover_libvirtd(params,
+                               remote_host=False,
+                               extra_params=None,
+                               is_recover=False,
+                               libvirtd_config=None):
+    """
+    Configure /etc/libvirt/libvirtd.conf on local and remote host if needed
+
+    :param params: A dict used to configure libvirtd.conf
+    :param remote_host True for setting up for remote host, too.
+                       False for not setting up
+    :param extra_params: Parameters for remote host
+    :param is_recover: True for recovering the libvirtd configuration and
+                               libvirtd_config should be provided
+                       False for configuring the libvirtd configuration.
+    :param libvirtd_config: an existing utils_config.LibvirtdConfig object
+    :return: utils_config.LibvirtdConfig object
+    """
+    server_ip = None
+    server_user = None
+    server_pwd = None
+
+    if remote_host:
+        server_ip = extra_params.get("server_ip")
+        server_user = extra_params.get("server_user")
```

Comment:
@dzhengfy `server_ip = extra_params.get("server_ip", "")` can be done instead of initializing to None?, same for rest

Reply:
@sathnaga sure, updated.

Reply:
@sathnaga I changed to configuration scope from libvirtd to libvirtd, qemu, sysconfig and guestconfig. Please help review again.